lib/dialects/oracle/connection-manager.js

```diff
@@ -0,0 +1,168 @@
+            this.lib.fetchAsString = sequelize.config.dialectOptions.fetchAsString;
+          } else {
+            this.lib.fetchAsString = [this.lib.CLOB];
+          }
+        }
+        this.lib.Promise = Promise;
+      }
+    } catch (err) {
+      if (err.code === 'MODULE_NOT_FOUND') {
+        throw new Error('Please install oracledb package manually');
+      }
+      throw err;
+    }
+
+  }
+
+  /**
+  * Method for checking the config object passed and generate the full database if not fully passed
+  * With dbName, host and port, it generates a string like this : 'host:port/dbname' 
+  */
+  checkConfigObject(config) {
+    //A connectString should be defined
+    if (config.database.length === 0) {
+      let errorToThrow = 'The database cannot be blank, you must specify the database name (which correspond to the service name';
+      errorToThrow += '\n from tnsnames.ora : (HOST = mymachine.example.com)(PORT = 1521)(SERVICE_NAME = orcl)'; 
+      throw new Error(errorToThrow);
+    }
+
+    if (!config.host || config.host.length === 0) {
+      throw new Error('You have to specify the host');
+    }
+
+    //The connectString has a special format, we check it
+    //ConnectString format is : host:[port]/service_name
+    if (config.database.indexOf('/') === - 1) {
+      
+      let connectString = config.host;
+
+      if (config.port && config.port !== 0) {
+        connectString += `:${config.port}`;
+      } else {
+        connectString += ':1521'; //Default port number
+      }
+      connectString += `/${config.database}`;
+      config.database = connectString;
+    }
+  }
+
+  // Expose this as a method so that the parsing may be updated when the user has added additional, custom types
+  _refreshTypeParser(dataType) {
+    parserStore.refresh(dataType);
+  }
+
+  _clearTypeParser() {
+    parserStore.clear();
+  }
+
+  connect(config) {
+    const connectionConfig = {
+      user: config.username,
+      host:  config.host,
+      port : config.port,
+      database : config.database,
+      password: config.password,
+      externalAuth : config.externalAuth,
+      stmtCacheSize : 0
+    };
+
+    //We check if there dialect options
+    if ('dialectOptions' in config) {
+      const dialectOptions = config.dialectOptions;
+
+      //If stmtCacheSize is defined, we set it
+      if (dialectOptions && 'stmtCacheSize' in dialectOptions) {
+        connectionConfig.stmtCacheSize = dialectOptions.stmtCacheSize;
+      }
+    }
+
+
+    //Check the config object
+    this.checkConfigObject(connectionConfig);
+
+    //We assume that the database has been correctly formed
+    connectionConfig.connectString = connectionConfig.database;
+
+    if (config.dialectOptions) {
+      Object.keys(config.dialectOptions).forEach(key => {
+        connectionConfig[key] = config.dialectOptions[key];
+      });
+    }
+
+    return this.lib.getConnection(connectionConfig).then(connection => {
+      return connection;
```

Comment:
:). then looks kinda unnecessary

Reply:
Agree, that should be removed. 

Also, the promise used here is generated the underlying lib, which defaults to the global Promise. If Sequelize expects all promises to be based on Utils.Promise, then this should be updated.

Reply:
Does `oracledb` library allow overriding what Promise instance is used ? this is essential to support cls support. Otherwise we need to manually override it everywhere we get result from library

Reply:
@sushantdhiman Yes, just [set oracledb.Promise](https://github.com/oracle/node-oracledb/blob/master/doc/api.md#-191-custom-promise-libraries) to the promise implementation you want to use.

Not sure where exactly this should be done in Sequelize.

Reply:
In connection-manager (this file's) constructor when setting `this.lib`, Also just to be safe, wrap `connect / disconnect` method with `Promise.try / new Promise` or whatever make sense.

Also wrap `connection.execute or _run` method in this Promise something like this https://github.com/sequelize/sequelize/blob/63978e184a95a8a39595cb0e04d43e663163c9bc/lib/dialects/mssql/query.js#L61

Reply:
@sushantdhiman Shouldn't that be a new Utils.Promise?

Reply:
@dmcghan It is, its imported as name `Promise`